bin/src/command/mod.rs

```diff
@@ -288,26 +288,28 @@ impl CommandServer {
             backends_count,
             frontends_count,
             accept_cancel: Some(accept_cancel_tx),
-        }
+        })
     }
 
-    pub fn disable_cloexec_before_upgrade(&mut self) {
+    pub fn disable_cloexec_before_upgrade(&mut self) -> anyhow::Result<()> {
         for ref mut worker in self.workers.iter_mut() {
             if worker.run_state == RunState::Running {
-                util::disable_close_on_exec(worker.fd);
+                util::disable_close_on_exec(worker.fd)?;
```

Comment:
there might be a case here where some workers have invalid file descriptors, while others don't, soe we should not quit the loop here

Reply:
How about logging the error instead?

Reply:
sure